docs/build_priority.rst

```diff
@@ -0,0 +1,100 @@
+Priority of Container Image Builds
+==================================
+
+For a build system it's desirable to prioritize different kinds of builds in
+order to better utilize resources. Unfortunately, OpenShift's scheduling
+algorithm does not support setting a priority value for a given build. To
+achieve some sort of build prioritization, we can leverage node selectors to
+allocate different resources to different build types.
+
+Consider the following types of container builds builds:
+
+- *scratch build*
+- *explicit build*
+- *auto rebuild*
+
+As the name implies, *scratch builds* are meant to be used as a one-off
+unofficial container build. No guarantees are made for storing the created
+container images long term. It’s also not meant to be shipped to customers.
+These are clearly low priority builds.
+
+*Explicit builds* are those triggered by a user, either directly via fedpkg/koji
+CLI, or indirectly via pungi (as in the case of base images). These are official
+builds that will go through the normal life cycle of being tested and,
+eventually, shipped.
+
+*Auto rebuilds* are created by OpenShift when a change in the parent image is
+detected. It’s likely that layered images should be rebuilt in order to pick up
+changes in latest parent image.
+
+For any *explicit build* or *auto rebuild*, they may or may not be high
+priority. In some cases, a build is high priority due to a security fix, for
+instance. In other cases, it could be due to an in-progress feature. For this
+reason, it cannot be said that all *explicit builds* are higher priority than
+*auto rebuilds*, or vice-versa.
+
+However, *auto rebuilds* will be enabled by a new feature that has the potential
+of completely consuming OSBS’s infrastructure. There must be some mechanism to
+throttle the amount of *auto rebuilds*. For this reason, OSBS uses a different
+node selector for each different build type:
+
+- *scratch build*: builds_scratch=true
+- *explicit build*: builds_explicit=true
+- *auto rebuild*: builds_auto=true
+
+By controlling each type of builds individually, OSBS will have the necessary
+control for adjusting its infrastructure.
+
+For example, consider an OpenShift cluster with 5 compute nodes:
+
+======  =================== ==================== ================
+Node    builds_scratch=true builds_explicit=true builds_auto=true
+======  =================== ==================== ================
+Node 1  ✔                   ✔                    ✔
+Node 2  ✗                   ✔                    ✗
+Node 3  ✗                   ✗                    ✔
+Node 4  ✗                   ✔                    ✔
+Node 5  ✗                   ✔                    ✔
+======  =================== ==================== ================
+
+In this case, *scratch builds* can be scheduled only on **Node 1**; *explicit
+builds* on any node except **Node 3**; and auto builds on any node except **Node
+2**.
+
+Orchestrator vs Worker Builds
```

Comment:
maybe we can somehow make clear that this is just explanation why we've chosen workers only approach ?